ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/context/InputContextManager.java

```diff
@@ -77,9 +79,11 @@ public void dispose() {
 	 * Saves dirty contexts.
 	 */
 	public void save(IProgressMonitor monitor) {
-		for (InputContext context : inputContexts.values()) {
+		Collection<InputContext> values = inputContexts.values();
+		SubMonitor subMon = SubMonitor.convert(monitor, values.size());
+		for (InputContext context : values) {
 			if (context.mustSave())
-				context.doSave(monitor);
+				context.doSave(subMon.newChild(1));
```

Comment:
```suggestion
				context.doSave(subMon.split(1));
```

Reply:
split() would check for cancel. i don't know if that is safe at this place and don't think it's needed - never had a lag there that i would like to cancel.